### ISSUE TITLE
PSCX 'System.Messaging' file not found on macOS

### ISSUE THREAD

**Issue by mikemaccana** [Pscx/Pscx#59]
On a clean install of macOS, PSCX will fails to load with the following error:

    ~/Documents $ Import-Module Pscx
    Import-Module : Could not load file or assembly 'System.Messaging, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
    At line:1 char:1
    + Import-Module Pscx
    + ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Import-Module], FileNotFoundException
    + FullyQualifiedErrorId : System.IO.FileNotFoundException,Microsoft.PowerShell.Commands.ImportModuleCommand

System details below:

    Name                           Value
    ----                           -----
    PSVersion                      6.2.0-preview.1
    PSEdition                      Core
    GitCommitId                    6.2.0-preview.1
    OS                             Darwin 17.7.0 Darwin Kernel Version 17.7.0: T...
    Platform                       Unix
    PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
    PSRemotingProtocolVersion      2.3
    SerializationVersion           1.1.0.1
    WSManStackVersion              3.0

**Comment by mikemaccana** [comment]
Ideally things like `Clear-MSMQueue` should be moved to a separate module (`PSCX-windows`) and the main `PSCX` work on any OS.

**Comment by sizur** [comment]
There are a lot of modules depending on Pscx and can't be imported on MacOS.  What's the workaround?

**Comment by tomoh1r** [comment]
PowerShell 7.0.0-rc.2 on CentOS 7 too...

**Comment by ysmoradi** [comment]
/-:

**Comment by jmamma** [comment]
Any workaround for this ? Same issue on Ubuntu.

**Comment by rkeithhill** [comment]
Yeah, the `MS` part of `MSQueue` means those commands are Windows-specific.  I'll look into conditionally importing Windows-specific commands only on Windows as an interim solution.  Should be easier to test now that I have Ubuntu and macOS systems to test on.

**Comment by painavin** [comment]
Any update on this issue?

**Comment by AlyaKoni** [comment]
6 years and no plan?

**Comment by hrxn** [comment]
https://github.com/Pscx/Pscx has not been updated for a long time now..

**Comment by oising** [comment]
> https://github.com/Pscx/Pscx has not been updated for a long time now..

It was created about 18 years ago. Eighteen. 

It's time for it to go out on its own. PRs are welcome.
